guides/micronaut-intellij-idea-ide-setup/micronaut-intellij-idea-ide-setup.adoc

```diff
@@ -0,0 +1,82 @@
+* Integration with the Bean Validation and Endpoints tool windows.
+* A dedicated project creation wizard based on launch.micronaut.io.
+* A dedicated run configuration for Micronaut applications.
+____
+
+:exclude-for-build:maven
+
+== IntelliJ IDEA and Gradle
+
+You can build Micronaut Framework with https://gradle.org/[Gradle]. The https://micronaut-projects.github.io/micronaut-gradle-plugin/latest/[Micronaut Gradle Plugin] offers a great development experience.
+
+https://www.jetbrains.com/help/idea/gradle.html[IntelliJ IDEA supports Gradle] out of the box both in the community and ultimate editions.
+
+== Build and run actions
+
+https://www.jetbrains.com/help/idea/work-with-gradle-projects.html#delegate_build_gradle[Delegate run and build actions to Gradle]:
+
+image::build-and-run-using-gradle.png[]
+
+____
+If you use the annotationProcessors in your project, we recommend that you delegate run and build actions to Gradle, so the annotation processor will be correctly enabled in your project.
+
+* In the Settings dialog (⌘ ,), go to Build, Execution, Deployment | Gradle.
+* On the Gradle settings page, in the Gradle Projects section, select a Gradle project you need.
+* In the Build and run using list, select the appropriate option and click OK to save the changes.
+____
+
+NOTE: Some features such as https://micronaut-projects.github.io/micronaut-test-resources/latest/guide/[test resources] rely on the Micronaut Gradle Plugin being used to run and test your application.
+
+:exclude-for-languages:groovy
+
+=== Enable annotation Processing
+
+If you don't delegate build and run actions to Gradle, make sure to enable annotation processing for Java and Kotlin applications
+
+image::annotationprocessorsintellij.png[]
+
+:exclude-for-languages:
+
+:exclude-for-build:
+
+:exclude-for-build:gradle
+
+== IntelliJ IDEA and Maven
+
+You can build Micronaut Framework with https://maven.apache.org/[Maven]. The https://micronaut-projects.github.io/micronaut-maven-plugin/latest/[Micronaut Maven Plugin] offers a great development experience.
+
+https://www.jetbrains.com/help/idea/maven-support.html[IntelliJ IDEA supports Maven] out of the box both in the community and ultimate editions.
+
+===  Build and run actions
+
+https://www.jetbrains.com/help/idea/delegate-build-and-run-actions-to-maven.html#delegate_to_maven[Delegate build and run actions to Maven]
+
+image::maven_settings_delegate.png[]
+____
+Build a project with Maven
+
+* Click Maven settings in the Maven tool window. Alternatively, from the main menu select File | Settings | Build, Execution, Deployment |Build Tools | Maven.
+
+* Click Maven and from the list, select Runner.
+
+* On the Runner page, select Delegate IDE build/run actions to maven.
+____
+
+NOTE: Some features such as https://micronaut-projects.github.io/micronaut-test-resources/latest/guide/[test resources] rely on the Micronaut Maven Plugin being used to run and test your application.
```

Comment:
I would put more emphasis on this, since it's the key reason to delegate to Maven/Gradle.

Generally speaking, the IntelliJ compiler will offer a better experience since it's truly incremental, and it's way faster than running the whole lifecycle of a Maven build.

Reply:
resolved via https://github.com/micronaut-projects/micronaut-guides/pull/1241/commits/2522ad324c6816cfffbea3df5e196aaf26bab7f9